core/blockchain/abci/abci.go

```diff
@@ -194,6 +195,163 @@ func AddCommonDeliverTxEvents(resp types.ResponseDeliverTx, tx Tx) types.Respons
+	TxTypeTransfer                       = "Transfer"
+	TxTypeOrderSubmission                = "OrderSubmission"
+	TxTypeOrderCancellation              = "OrderCancellation"
+	TxTypeOrderAmendment                 = "OrderAmendment"
+	TxTypeVoteSubmission                 = "VoteSubmission"
+	TxTypeWithdrawSubmission             = "WithdrawSubmission"
+	TxTypeLiquidityProvisionSubmission   = "LiquidityProvisionSubmission"
+	TxTypeLiquidityProvisionCancellation = "LiquidityProvisionCancellation"
+	TxTypeProposalSubmission             = "ProposalSubmission"
+	TxTypeAnnounceNode                   = "AnnounceNode"
+	TxTypeNodeVote                       = "NodeVote"
+	TxTypeNodeSignature                  = "NodeSignature"
+	TxTypeOracleDataSubmission           = "OracleDataSubmission"
+	TxTypeDelegateSubmission             = "DelegateSubmission"
+	TxTypeUndelegateSubmission           = "UndelegateSubmission"
+	TxTypeKeyRotateSubmission            = "KeyRotateSubmission"
+	TxTypeStateVariableProposal          = "StateVariableProposal"
+	TxTypeCancelTransfer                 = "CancelTransfer"
+	// TxTypeValidatorHeartbeat                 = "ValidatorHeartbeat".
+	TxTypeEthereumKeyRotateSubmission = "EthereumKeyRotateSubmission"
+	TxTypeProtocolUpgradeProposal     = "ProtocolUpgradeProposal"
+	TxTypeIssueSignatures             = "IssueSignatures"
+	TxTypeBatchMarketInstructions     = "BatchMarketInstructions"
+	TxTypeERC20MultisigSignerAdded    = "ERC20MultisigSignerAdded"
+	TxTypeERC20MultisigSignerRemoved  = "ERC20MultisigSignerRemoved"
+)
+
+type TxType struct {
+	Type     string
+	Sender   string
+	Receiver string
+}
+
+// nolint: nakedret
+func GetTxType(tx Tx) (txt TxType) {
+	switch c := tx.GetCmd().(type) {
+	case *commandspb.ChainEvent:
+		if se := c.GetStakingEvent(); se != nil {
+			if sed := se.GetStakeDeposited(); sed != nil {
+				txt.Type = TxTypeStakeDeposited
+				txt.Receiver = sed.GetVegaPublicKey()
+			}
+			if ser := se.GetStakeRemoved(); ser != nil {
+				txt.Type = TxTypeStakeRemoved
+				txt.Receiver = ser.GetVegaPublicKey()
+			}
+		}
+		if bi := c.GetBuiltin(); bi != nil {
+			if bid := bi.GetDeposit(); bid != nil {
+				txt.Type = TxTypeBuiltinAssetDeposit
+				txt.Receiver = bid.GetPartyId()
+			}
+			if biw := bi.GetWithdrawal(); biw != nil {
+				txt.Type = TxTypeBuiltinAssetWithdrawal
+				txt.Sender = biw.GetPartyId()
+			}
+		}
+		if e20 := c.GetErc20(); e20 != nil {
+			if e20d := e20.GetDeposit(); e20d != nil {
+				txt.Type = TxTypeERC20Deposit
+				txt.Receiver = e20d.GetTargetPartyId()
+			}
+			if e20w := e20.GetWithdrawal(); e20w != nil {
+				txt.Type = TxTypeERC20Withdrawal
+				txt.Sender = tx.Party() // TODO: what else to do here?
+			}
+		}
+		if erc20ms := c.GetErc20Multisig(); erc20ms != nil {
+			if erc20msadd := erc20ms.GetSignerAdded(); erc20msadd != nil {
+				txt.Type = TxTypeERC20MultisigSignerAdded
+				txt.Sender = erc20msadd.GetNewSigner() // TODO: what else to do here?
```

Comment:
I guess this doesn't really make a huge amount of sense semantically but if it's something we want to be able to filter on in the block explorer it's not the worst hack; without adding attributes for just about every field any sort of tx can have

Reply:
yeah that was my thinking. We will end up having too many attributes recorded. We can always document it nicely